### PR TITLE
final project @rickyc

### PR DESCRIPTION
**Pull request by erinleeny**
I can’t find the bug that keeps tripping my code :(

### CODE REVIEW

**Comment by rickyc** [comment]
What's the 🐛?

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+GOOGLE_INIT = "1tOatwbeCtqw67v7dE1s-FjE7SOJQ-qxthn04nMnauFY"
```
Comment (rickyc) [inline .env:1]: 😱 please reset this key

File: bluehex.html [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+<link rel="stylesheet" type="text/css"
+href="https://fonts.googleapis.com/css?family=Pacifico">
+<link rel="stylesheet" type="text/css"
+href="https://fonts.googleapis.com/css?family=Amatic+SC">
+
+</head>
+<body>
+  <div class = "container">
+    <div class = "row">
+      <center><div style="font-family: 'Pacifico', serif; font-size: 60px;">Complete the day</div>
```
Comment (rickyc) [inline bluehex.html]: You should try not to use inline css. Assign it a class or ID and style this instead. 

```css
text-align: center;
```

is how you center things without the `<center>` tag

File: bluehex.html [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+            <br>
+            <div class = "text">
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </body>
+  <script>
+  var x = 0;
```
Comment (rickyc) [inline bluehex.html]:
```js
let x = 0;
```

Comment (rickyc) [reply bluehex.html]: Wait, why is this even here? What is this for? I noticed your `for` loop used `x` as well.

File: bluehex.html [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+    $("#wrapper").append("<div class='row" + i + "'>");
+    for(let x = 0; x < 10; x++){
+      $("#wrapper").append("<div class = 'box' id='" + i + "-" + x + "'></div>")
+    }
+  }
+  let column = 0;
+  let row = 0;
+  //random blocks
+  for(let i = 0; i <= 25; i++){
+    let rand = Math.floor(Math.random() * (10)) + 0;
```
Comment (rickyc) [inline bluehex.html]: Notice how many times you are repeating this line of code. What did I say you should do when you are repeating code? 🤔

File: bluehex.html [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+    let rand2 = Math.floor(Math.random() * (10)) + 0;
+    $("#" + rand + "-" + rand2).addClass('block');
```
Comment (rickyc) [inline bluehex.html]: Do you notice that you are writing `$("#" + rand + "-" + rand2)` this part a lot. Again, what can you do?

File: bluehex.html [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+  for(let i = 0; i <= 11; i++){
+    let rand = Math.floor(Math.random() * (10)) + 0;
+    let rand2 = Math.floor(Math.random() * (10)) + 0;
+    $("#" + rand + "-" + rand2).attr("style", `background: ${array[i]}`).addClass('energy').removeClass('block');
+    console.log(rand + " " + rand2);
+  }
+  $('#' + row + '-' + column).addClass('player').removeClass('block');
+  arrayNum = 0
+  $("#scoreboard").append("<br><br><div><h2>color you need: </h2><div class = 'box colorNeed' style = 'background-color:" + array[arrayNum] +"'</div></div>");
+  function energyCollect (arrayNum, array, row, column){
```
Comment (rickyc) [inline bluehex.html]: Indentation needs to be fixed.

File: bluehex.html [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+  console.log(arrayNum);
+  arrayNum = arrayNum + 1;
+  console.log(arrayNum);
+  $(".box.colorNeed").attr("style", `background: ${array[arrayNum]}`);
+  $('#' + row + '-' + column).removeClass('energy').attr("style", `background: black`);
+  }
+  document.onkeydown = function myFunction() {
```
Comment (rickyc) [inline bluehex.html]: This is okay, I will allow it, but the jQuery way is.


```js
$(document).keydown(() => {
  // https://api.jquery.com/keydown/
});
```

File: bluehex.html [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+    switch (event.keyCode) {
+      //up key
+      case 38:
+      //checks if player is on top row
```
Comment (rickyc) [inline bluehex.html]: Notice how you have a logic in this case statement. You can group the code into a function.

```
function upArrowToggled() {
  // does stuff
}
```

**Review by rickyc — COMMENTED** [review]
Just leaving some feedback. There are plenty of refactoring you can do with the code. I think that will make it easier to debug once you start grouping code into functions.

**Comment by erinleeny** [comment]
@rickyc sometimes there'll be a color that isnt shown on the board, and when i collect the correct color, it doesnt respond. This happens after the 2nd or 3rd time, though.

**Comment by rickyc** [comment]
My suggestion would be to log the colors as you move across the board. Right now there are no debugger statements (console.log), it is difficult to track down the problem.

For example things to log would be, what is my current player color, what is the color of the tile I am on. With this when you step on the color, you can see if it actually matches. Without any debuggers you are basically just blindly guessing.

**Pull request merged**